plugins/sync.js

```diff
@@ -83,73 +84,93 @@ window.plugin.sync.RegisteredMap = function(options) {
+                                                            // 'no' sync needed
+                                                            // force 'remote' (1st sync)
+                                                            // force 'local'
+                                                            // sync is on 'stop'
+  
   this.initialize = this.initialize.bind(this);
   this.loadDocument = this.loadDocument.bind(this);
 };
 
+// this function updates the registered map (copy of the plugin's data)
+// and finally saves it to google drive
+// it includes this clients UUID so that a reading client can see if 
+// if an other clinet did the last update or if it was this client
 window.plugin.sync.RegisteredMap.prototype.updateMap = function(keyArray) {
   var _this = this;
   try {
-    this.lastUpdateUUID = this.uuid;
-
-    $.each(keyArray, function(ind, key) {
-      var value = window.plugin[_this.pluginName][_this.fieldName][key];
-      if(typeof(value) !== 'undefined') {
-        _this.map[key] = value;
-      } else {
-        delete _this.map[key];
+    this.lastUpdateUUID = this.uuid;                        // use this clients uuid
+    $.each(keyArray, function(ind, key) {                   // find all changed elements from the array
+      var value = window.plugin[_this.pluginName][_this.fieldName][key]; 
+      if(typeof(value) !== 'undefined') {                  
+        _this.map[key] = value;                             // replace element in the array
+      } else {                                              // or
+        delete _this.map[key];                              // remove it from the array
       }
     });
   } finally {
-    _this.dataStorage.saveFile(_this.prepareFileData());
+    _this.dataStorage.saveFile(_this.prepareFileData());    // save data and UUID to drive
   }
 };
 
-window.plugin.sync.RegisteredMap.prototype.isUpdatedByOthers = function() {
-  var remoteUUID = this.lastUpdateUUID.toString();
-  return (remoteUUID !== '') && (remoteUUID !== this.uuid);
-};
-
+// concatenate a the filename for the drive "pluginName[fieldName]"
 window.plugin.sync.RegisteredMap.prototype.getFileName = function() {
   return this.pluginName + '[' + this.fieldName + ']'
 };
 
+// initialize access to the sync-file on drive
+//   callback (optional) - to be performed on success
 window.plugin.sync.RegisteredMap.prototype.initFile = function(callback) {
   var assignIdCallback, failedCallback, _this;
-  _this = this;
+  _this = this;                                             // have _this point to this Object
 
-  assignIdCallback = function(id) {
+  // file access granted, perform callback-function
+  assignIdCallback = function(id) {                         // where does this id come from?
     _this.forceFileSearch = false;
-    if(callback) callback();
+    if(callback) callback();                                // do the callback passed to .initfile
```

Comment:
```suggestion
    if (callback) callback();
```
maybe rename "callback" to "onSuccessCallback" ?